src/coders/byte-stream.js

```diff
@@ -1,71 +1,137 @@
+     * stream before calling readStruct.
+     */
     readStruct (StructType) {
-        const obj = new StructType(this.uint8, this.position);
+        const obj = new StructType(this.uint8a, this.position);
         this.position += StructType.size;
         return obj;
     }
 
+    /**
+     * Resize the internal buffer to allow for the needed amount of space.
+     * @param {number} needed - How many bytes need to fit in the buffer.
+     * @private
+     */
     resize (needed) {
         if (this.buffer.byteLength < needed) {
-            const uint8 = this.uint8;
+            const uint8a = this.uint8a;
             const nextPowerOf2 = Math.pow(2, Math.ceil(Math.log(needed) / Math.log(2)));
             this.buffer = new ArrayBuffer(nextPowerOf2);
 
-            this.uint8 = new Uint8Array(this.buffer);
-            this.uint8.set(uint8);
+            this.uint8a = new Uint8Array(this.buffer);
+            this.uint8a.set(uint8a);
         }
     }
 
+    /**
+     * Write a value to the stream with a BytePrimitive defining how to do so
+     * and increment the position.
+     * @param {BytePrimitive} member - BytePrimitive to define how to write the
+     * value.
+     * @param {*} value - Value to write.
+     * @returns {*} Value passed to the method.
+     */
     write (member, value) {
         if (member.size === 0) {
             this.resize(this.position + member.writeSizeOf(value));
         } else {
             this.resize(this.position + member.size);
         }
 
-        member.write(this.uint8, this.position, value);
+        member.write(this.uint8a, this.position, value);
         if (member.size === 0) {
-            this.position += member.writeSizeOf(this.uint8, this.position);
+            this.position += member.writeSizeOf(this.uint8a, this.position);
         } else {
             this.position += member.size;
         }
         return value;
     }
 
+    /**
+     * Write data to the stream structured by a given Packet subclass
+     * constructor and increment the position.
+     * @param {PacketConstructor} StructType - The Packet subclass constructor
+     * defining how to write the data.
+     * @param {object} data - Data to write.
+     * @returns {Packet} - Constructed packet after writing data.
+     */
     writeStruct (StructType, data) {
         this.resize(this.position + StructType.size);
 
-        const st = Object.assign(new StructType(this.uint8, this.position), data);
+        const st = Object.assign(new StructType(this.uint8a, this.position), data);
         this.position += StructType.size;
         return st;
     }
 
+    /**
+     * Write bytes from the given Uint8Array array to the stream and increment
+     * the position.
+     * @param {Uint8Array} bytes - Bytes to write to the stream.
+     * @param {number=} [start=0] - Where in bytes to start writing from.
+     * @param {nubmmer=} [end=bytes.length] - Where in bytes to stop writing.
```

Comment:
Maybe add whether bytes[end] will be written to or not. It looks like not..

Also there is a typo: `nubmmer` => `number`

```suggestion
      * @param {number=} [end=bytes.length] - Where in bytes to stop writing, excluding position at bytes[end].
```